### ISSUE TITLE
Very gradient-intensive domain

### ISSUE THREAD

**Issue by GLivshits** [eladrich/pixel2style2pixel#175]
Hi. I'm using your approach to invert a generator for fingerprints (256 resolution, some of them are or cartoon-type, grayscale). The generator works perfectly, but inversion is quite tough. For a long time, it outputs just a blurry image of the finger (about 10000 iters). But then, suddenly, L2 loss explodes (from the value of 0.02 for 0.-1. image data up to 20K-11M).
Here are examples before and after the explosion:
![photo_2021-07-20_18-31-26](https://user-images.githubusercontent.com/47557983/126352692-224034e1-a137-40c4-b792-892b1f47f867.jpg)
![photo_2021-07-20_18-31-29](https://user-images.githubusercontent.com/47557983/126352716-27f24cf4-ba0a-4ab2-9d56-51837e7d5fef.jpg)
Moreover, Adam performing much better than Ranger. 
How to deal with it??? I blame the highly abruptive image gradients, which are characteristic of fingerprint images.

**Comment by yuval-alaluf** [comment]
This is definitely a challenging domain. Maybe you do need to play with the optimizer parameters. 
What I typically like to do is to try to overfit on one or a few examples to make sure that my model even has the capacity to learn what I want it to learn. Only after I successfully overfit something so I move on to training on a larger data set. I think such an approach could help you here if you haven't done so already.

**Comment by GLivshits** [comment]
Oh, this is very simple and yet amazing idea. Thanks, I'll try to do so.

**Comment by mmitchef** [comment]
Hi @GLivshits could you get this code to work on your model?
